### PR TITLE
Update README

### LINKED ISSUE #49: Update README.md featuring plugin support
- Feature custom prompt support
- Add README.md for each plugin
- README.md for plugin setup and development

@frmendes I'm creating this just to keep track of it.

### PR DESCRIPTION
**Pull request by frm**
Update the README with information about new architecture, features and missing screenshots.

There's a generic README on the root of the project, one on the plugins directory, detailing how plugins work and finally one per plugin.

Feel free to make corrections @desyncr

Should fix #49

### CODE REVIEW

**Comment by desyncr** [comment]
Looks good. I'm take a closer look and if everything is OK I'm gonna squash everything and put it live!

**Comment by frm** [comment]
Please do. I probably missed something. Long day.

File: README.md [hunk 1]
```diff
@@ -40,64 +31,144 @@ set `ZSH_THEME="geometry"` in your `.zshrc`.
...
+- display the running time of long running commands
+- set the terminal title to current command and directory
+- make you the coolest hacker in the whole Starbucks
+
+The right side prompt is print asynchronously, so you know it's going to be
+fast™.
+
+## Plugins
+
+geometry has an internal plugin architecture. The default plugins are `exec_time` and `git`. But you can also enable a number of them just by setting the `GEOMETRY_PROMPT_PLUGINS` variable in your own configuration files:
```
Comment (desyncr) [inline README.md]: The variable for built-in plugins is `GEOMETRY_PROMPT_DEFAULT_PROMPTS`, which, looking at it now, is a terrible name.

This is different from `GEOMETRY_PROMPT_PLUGINS` due that in the setup phrase we source plugins from listed in `GEOMETRY_PROMPT_DEFAULT_PROMPTS`; we use `GEOMETRY_PROMPT_PLUGINS` to keep track of registered plugins.

I guess we should rename `GEOMETRY_PROMPT_DEFAULT_PROMPTS` to `GEOMETRY_PROMPT_PLUGINS` and `GEOMETRY_PROMPT_PLUGINS` to `_GEOMETRY_PROMPT_PLUGINS` as it's actually a sort of private variable.

File: README.md [hunk 2]
```diff
@@ -40,64 +31,144 @@ set `ZSH_THEME="geometry"` in your `.zshrc`.
...
+
+```sh
+GEOMETRY_PROMPT_PLUGINS=(virtualenv dockermachine)
```
Comment (desyncr) [inline README.md]: `dockermachine` plugin is actually `docker_machine`.

File: plugins/README.md [hunk 3]
```diff
@@ -1,5 +1,71 @@
...
+
+*Note: if you're not sure where to put geometry configs, just add them to your `.zshrc`*
+
+## Custom plugins
+
+If you want to set up your own custom plugin, it's pretty straightforward to do
+so. All you need is a `setup` and a `render` function, with the plugin name on
+them.
+
+Let's assume you want to add a plugin that prints `(☞ﾟ∀ﾟ)☞` when the git branch
```
Comment (desyncr) [inline plugins/README.md:33]: Is this plugin available? I actually want it :-D

**Review by desyncr — CHANGES_REQUESTED** [review]
There are a few things to check out. Overall a great work!

File: README.md (hunk 1 above)
Comment (desyncr) [reply README.md]: I'm gonna take a look at this [here](https://github.com/frmendes/geometry/issues/55).

File: plugins/README.md [hunk 4]
```diff
@@ -1,5 +1,71 @@
...
+* [Virtualenv](/plugins/virtualenv)
+
+## Default plugins
+
+By default, geometry uses `git` and `exec_time`. You can add more by changing
+the `GEOMETRY_PROMPT_PLUGINS` variable in your own configuration files.
+
+
+```sh
+GEOMETRY_PROMPT_PLUGINS=(virtualenv dockermachine)
```
Comment (desyncr) [inline plugins/README.md]: As this variable is not defined until geometry is loaded, it's empty as this time, so we can't _add_ plugins to the defaults, but _define_ the defaults. So as an example I'd put:

    GEOMETRY_PROMPT_PLUGINS=(virtualenv docker_machine exec_time git)

So we make it clear, and I'll change the above paragraph for the following: 

    By default, geometry uses `git` and `exec_time`. You can configure a different setup by changing the `GEOMETRY_PROMPT_PLUGINS` variable in your own configuration files.

File: plugins/README.md [hunk 5]
```diff
@@ -1,5 +1,71 @@
...
+*Note: if you're not sure where to put geometry configs, just add them to your `.zshrc`*
+
+## Custom plugins
+
+If you want to set up your own custom plugin, it's pretty straightforward to do
+so. All you need is a `setup` and a `render` function, with the plugin name on
+them.
+
+Let's assume you want to add a plugin that prints `(☞ﾟ∀ﾟ)☞` when the git branch
+is clean and `(ノಠ益ಠ)ノ彡┻━┻` when it's dirty. Well call it `pretty_git`.
```
Comment (desyncr) [inline plugins/README.md]: "Well call it `pretty_git`." -> "Let's call it `pretty_git`."

File: plugins/docker_machine/README.md [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+
+### Colors
+
+You can change the color by setting the `GEOMETRY_COLOR_DOCKER_MACHINE`
+environment variable. Defaults to `blue`.
+
+### Symbols
+
+You can change the symbol by setting the `GEOMETRY_SYMBOL_DOCKER_MACHINE`
+environment variable. Defaults to `⚓.
```
Comment (desyncr) [inline plugins/docker_machine/README.md]: There is an unclosed backtick.

**Review by desyncr — CHANGES_REQUESTED** [review]
A few more small things to look for.

**Comment by desyncr** [comment]
Most things in review are small things, let me know if you want me to go ahead and fix them up. Once #55 is merged we can move on and get it live!

**Pull request merged**